### DISCUSSION TITLE
是否支持加载外部训练的模型进行相似度量和相似搜索

### DISCUSSION THREAD

**Discussion by walkerrbytes** [milvus-io/milvus#33113] [Q&A and General discussion] [answered]
我们已经拥有了一个训练好的pt模型文件，能否在创建库表时加载作为外部的Similarity Metrics。然后进行相似度量和相似搜索

**Comment by yhmo** [comment, marked as answer]
目前还不支持自定义的相似度量，没有这样的接口。见过有少数用户提过类似的需求，不过要实现恐怕也是3.0版本之后的事情了。

**Reply by walkerrbytes** [reply]
好的，感谢回复

**Comment by xiaofan-luan** [comment]
> 我们已经拥有了一个训练好的pt模型文件，能否在创建库表时加载作为外部的Similarity Metrics。然后进行相似度量和相似搜索

请问你们模型使用什么距离计算公式呢？有什么具体原因没有用更加通用的么，方便透露便于我们解决么

**Reply by walkerrbytes** [reply]
目前情况是。我们有一个训练出的预训练权重load到模型中，模型继承自nn.Module的类来完成的，这个类定义了模型的结构以及自定义了一些embedding之间的操作当作距离，并作为forward方法返回。

**Comment by xiaofan-luan** [comment]
I've seen this a couple of times. 
What will be idea way to offer the customized function? load a dynamic lib?

**Comment by xiaofan-luan** [comment]
how heavy is each calculation?
